core/src/main/java/org/jdbi/v3/core/mapper/reflect/ConstructorInstanceFactory.java

```diff
@@ -22,9 +28,18 @@
 class ConstructorInstanceFactory<T> extends InstanceFactory<T> {
     private final Constructor<T> constructor;
 
+    private final Supplier<Type[]> typeSupplier;
+
     ConstructorInstanceFactory(Constructor<T> constructor) {
         super(constructor);
         this.constructor = requireNonNull(constructor, "constructor is null");
+        this.typeSupplier = isRecordConstructor(constructor) ? getRecordParameterTypeSupplier(constructor) : super::getTypes;
```

Comment:
was thinking to use inheritance but then this method would have to be hoisted up, more code would change, but I can do that if it's preferred.

Reply:
I think keeping the change small is good - unless I misunderstand, it would be nice to remove this code once the upstream bug is fixed.